doc/sailfish-webview-qmozenginesettings.cpp

```diff
@@ -0,0 +1,219 @@
+
+    \sa {SailfishOS::WebView::initialize}{WebView::initialize}
+    \sa {SailfishOS::QMozContext::initialize}{QMozContext::initialize}
+*/
+
+/*!
+    \property QMozEngineSettings::autoLoadImages
+    \brief Whether to automatically load images.
+
+    When set to \c true images will be automatically downloaded and rendered.
+    When set to \c false images will not be downloaded.
+
+    The default value is \c true.
+
+    This corresponds to the "permissions.default.image" gecko preference.
+
+    Setting this property to \c true can help avoid tracking by external
+    servers when rendering local content, but may prevent a page from rendering
+    correctly.
+*/
+
+/*!
+    \property QMozEngineSettings::javascriptEnabled
+    \brief Whether javascript on the page will be automatically executed.
+
+    When set to \c true javascript on the page will be executed. When set to
+    \c false javascript will not be executed.
+
+    The default value is \c true.
+
+    This corresponds to the "javascript.enabled" gecko preference.
+
+    Disabling javascript may be beneficial for security, privacy or efficiency
+    reasons, but may also prevent pages from rendering correctly.
+*/
+
+/*!
+    \property QMozEngineSettings::popupEnabled
+    \brief Whether to allow popups to open during page loading.
+
+    When set to \c true any popups the page requests to open during loading will
+    be allowed to open. When set to \c false requests for popups will be
+    blocked from being actioned.
+
+    The default value is \c false.
+
+    This corresponds to the "dom.disable_open_during_load" gecko preference.
+*/
+
+/*!
+    \enum QMozEngineSettings::CookieBehavior
+
+    This enum type specifies the cookie behaviour for the engine to use.
+
+    \value AcceptAll
+           Accept all cookies independent of their origin.
+    \value BlockThirdParty
+           Accept cookies only if they are from the requested domain.
+    \value BlockAll
+           Do not accept any cookies.
+    \omitvalue Deprecated
+*/
+
+/*!
+    \property QMozEngineSettings::cookieBehavior
+    \brief Sets the cookie behaviour.
+
+    The cookie behaviour can be one of:
+
+    \value QMozEngineSettings.AcceptAll
+           Accept all cookies independent of their origin, the default
+    \value QMozEngineSettings.BlockThirdParty
+           Accept cookies only if they are from the requested domain.
+    \value QMozEngineSettings.BlockAll
+           Do not accept any cookies.
+
+    This corresponds to the "network.cookie.cookieBehavior" gecko preference.
+
+    \sa QMozEngineSettings::CookieBehavior
+*/
+
+/*!
+    \property QMozEngineSettings::useDownloadDir
+    \brief Sets whether or not to use the default download location.
+
+    When set to \c true downloaded files will be saved to the default download
+    location, as specified by the \l downloadDir property. When set to \c false
+    the user will be asked to specify a location for each file downloaded.
```

Comment:
Out of interest, how will they be asked to specify a location?  Is it a popup?  which async message?

Reply:
Yes, it's using a `FolderPicker` popup. The message is `"embed:downloadpicker"` and it triggers [this bit of code](https://github.com/sailfishos/sailfish-components-webview/blob/master/import/pickers/PickerOpener.qml#L69). Do you think there should be something added to the docs about this?

Maybe it'd be good to briefly document all the relevant EmbedLite messages somewhere.